azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool.go

```diff
@@ -0,0 +1,321 @@
+						"expiration_time": {
+							Type:         schema.TypeString,
+							ValidateFunc: validation.IsRFC3339Time,
+							Required:     true,
+						},
+
+						"reset_token": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"token": {
+							Type:      schema.TypeString,
+							Sensitive: true,
+							Computed:  true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceArmVirtualDesktopHostPoolCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DesktopVirtualization.HostPoolsClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Virtual Desktop Host Pool creation")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Virtual Desktop Host Pool %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_virtual_host_pool", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	t := d.Get("tags").(map[string]interface{})
+
+	hostPoolType := d.Get("type").(string)
+	friendlyName := d.Get("friendly_name").(string)
+	description := d.Get("description").(string)
+	ValidationEnvironment := d.Get("validation_environment").(bool)
+	LoadBalancerType := d.Get("load_balancer_type").(string)
+	PersonalDesktopAssignmentType := d.Get("personal_desktop_assignment_type").(string)
+	PreferredAppGroupType := d.Get("preferred_app_group_type").(string)
+	maxSessionLimit := int32(d.Get("max_session_limit").(int))
+
+	context := desktopvirtualization.HostPool{
+		Location: &location,
+		Tags:     tags.Expand(t),
+		HostPoolProperties: &desktopvirtualization.HostPoolProperties{
+			HostPoolType:                  desktopvirtualization.HostPoolType(hostPoolType),
+			FriendlyName:                  &friendlyName,
+			Description:                   &description,
+			ValidationEnvironment:         &ValidationEnvironment,
+			MaxSessionLimit:               &maxSessionLimit,
+			LoadBalancerType:              desktopvirtualization.LoadBalancerType(LoadBalancerType),
+			PersonalDesktopAssignmentType: desktopvirtualization.PersonalDesktopAssignmentType(PersonalDesktopAssignmentType),
+			PreferredAppGroupType:         desktopvirtualization.PreferredAppGroupType(PreferredAppGroupType),
+			RegistrationInfo:              expandVirtualDesktopHostPoolRegistrationInfo(d),
+		},
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resourceGroup, name, context)
+	if err != nil {
+		return fmt.Errorf("Error creating Virtual Desktop Host Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
```

Comment:
Error is redundant in these error messages
```suggestion
		return fmt.Errorf("creating Virtual Desktop Host Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
```